### PR TITLE
[FLINK-36076] Set isSchemaChangeApplying as isSchemaChangeApplying for thread safe.

### PR DESCRIPTION
**Pull request by loserwang1024**
As discussed in https://issues.apache.org/jira/browse/FLINK-36076 , Current, we apply schema change asynchronous.  
SchemaChangeThreadPool will set isSchemaChangeApplying as true when apply schema change event. And the main thread which handle operater event will check isSchemaChangeApplying.
 
For thread safe, this should be volatile.

### CODE REVIEW

**Comment by loserwang1024** [comment]
@leonardBang @PatrickRen @ruanhang1993 , CC

**Comment by leonardBang** [comment]
The PR title and commit content confuse me a little, @loserwang1024 could you check your PR ?

File: flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistryRequestHandler.java [hunk 1]
```diff
@@ -94,7 +94,7 @@ public class SchemaRegistryRequestHandler implements Closeable {
     private final Set<Integer> flushedSinkWriters;
 
     /** Status of the execution of current schema change request. */
-    private boolean isSchemaChangeApplying;
+    private transient boolean isSchemaChangeApplying;
```
Comment (leonardBang) [inline flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistryRequestHandler.java:97]: volatile?

**Review by leonardBang — APPROVED** [review]
Thanks for the update, +1

**Pull request merged**